resource_trigger/check.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+import requests
+import json
+import sys
+
+from utils import msg, parse_legacy_ids, determine_archive_server, BOOK_UUIDS
+
+
+def get_sync_file(token, repo):
+    headers = {'Authorization': 'token ' + token}
+    endpoint = f"https://raw.githubusercontent.com/openstax/{repo}/main/archive-syncfile"
+    resp = requests.get(endpoint, headers=headers)
+
+    if resp.status_code != 200:
+        msg(f"Error: Unable to get archive-syncfile for {repo}")
+        sys.exit(1)
+
+    sync_file = resp.content.decode()
+    return sync_file
+
+
+def _check(instream):
+    payload = json.load(instream)
+    source = payload['source']
+    token = source['github_token']
+    repo = source['book_repo']
+    server = source.get('archive_server')
+    archive_server = determine_archive_server(server)
+
+    msg(f"Syncing {repo} with {archive_server}...")
+
+    # headers = {'Authorization': 'token ' + token}
+    # endpoint = f"https://raw.githubusercontent.com/openstax/{repo}/main/archive-syncfile"
+    # resp = requests.get(endpoint, headers=headers)
+
+    # if resp.status_code != 200:
+    #     msg(f"Error: Unable to get archive-syncfile for {repo}")
+    #     sys.exit(1)
+
+    # sync_file = resp.content.decode()
+    sync_file = get_sync_file(token, repo)
+    legacy_ids = parse_legacy_ids(sync_file)
+
+    # Get versions from uuid via archive api
+    versions = []
+    for legacy_id in legacy_ids:
+        try:
+            uuid = BOOK_UUIDS[legacy_id]
+        except KeyError as error:
+            msg(error, file=sys.stderr)
+            sys.exit(1)
+        link = f"https://{archive_server}/contents/{uuid}"
```

Comment:
I'm pretty sure that this provides the `latest successfully baked` version and not the `latest` version of the collection. Have you seen something that indicates otherwise? For similar reasons, we do not use the `version` field of this API endpoint with neb to determine the latest version to pull for CMs. This may need to use the `extras` endpoint.

Reply:
ah interesting.. so when a 'publish' is done by neb, it doesn't mean it gets baked..?

Reply:
diff between - 
'latest successfully baked'
'latestVersion'
'headVersion'

example link: https://archive.cnx.org/extras/9d8df601-4f12-4ac1-8224-b450bf739e5f@6.10